### DISCUSSION TITLE
Question about main frame navigation raising `Ferrum::StatusError`

### DISCUSSION THREAD

**Discussion by francisbeaudoin** [rubycdp/ferrum#340] [General]
Hello,

I'm wondering if there's context behind only raising a `StatusError` for a subset of errors when navigating the top frame?

https://github.com/rubycdp/ferrum/blob/0cab51d1f0500efa7573cac39f99c44421f81c86/lib/ferrum/page.rb#L113-L124

Unless mistaken as only these 5 errors are handled, it implies that we need to inspect the network traffic to know whether navigation was successful. Should there be a lot of traffic, that is a bit cumbersome as the network traffic needs to be filtered to properly identify the main frame request (including its redirection) and then finally check if it has a `Exchange#error` to assess a successful top frame navigation.

**Comment by route** [comment]
Don't quite understand what are you proposing instead between the lines? What seems to be incorrect now and what the solution is.

**Comment by francisbeaudoin** [comment]
@route Sorry if that wasn't clear. 

I think that a `StatusError` should be raised if `Page.navigate` returns an `errorText`, regardless of the actual error code. 

Per the [documentation](https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-navigate) an `errorText` is present if and only if navigation has failed. Therefore, the proposition is to have something similar:
```
...
response = command("Page.navigate", wait: GOTO_WAIT, **options)
raise StatusError.new(options[:url], "Request to #{options[:url]} failed (#{response['errorText']}).") if response['errorText']
...
```

**Comment by route** [comment]
I see, the reason behind that `StatusError` is that in cuprite we'd like to show that page cannot be loaded https://github.com/rubycdp/cuprite/blob/7ace9f6292a7f7c4ebb5375077c067a0311cc4f1/spec/features/driver_spec.rb#L636 it stemmed from poltergeist. You are right I suppose and we can raise error in all cases. Mind sending a PR?
